vpn/tunnel.go

```diff
@@ -552,6 +554,42 @@ func (u *updater) netStatusLoop() {
 	}
 }
 
+// processFreshState handles the logic for when a fresh state update is received.
```

Comment:
You should probably expand this comment to explain why this is necessary. Should mention that we only receive diffs except for the first packet on any given reconnect to the tailnet API, which means that without this we weren't processing deletes for any workspaces or agents deleted while the client was disconnected (e.g. while the computer was asleep)